### ISSUE TITLE
[Bug] Type Annotation support for class type

### ISSUE THREAD

**Issue by ain-soph** [microsoft/pylance-release#824]
This issue is similar to #714 but not exactly the same. It should be fixed in #714, but it seems not.  

## Code:  
```python3
from typing import Type


class A:
    @staticmethod
    def func() -> float:
        return 1.0


# Works not well
b: dict[str, Type[A]] = {'a': A}
c = b['a']


number = c.func()
number.as_integer_ratio()  # But number type is float correctly, and it can lint as_integer_ratio without problem.


# Works well
a = A()
a.func()
A.func()

# Works well
d: dict[str, A] = {'a': a}
e = d['a']
e.func()

```
## Result: 
<img src="https://user-images.githubusercontent.com/13214530/104261397-73fe5a80-5453-11eb-8acc-8e3830017480.png" alt="image" width="200"/>

Even though the Pylance finds the `c.func` correctly (and has auto-complete. I think it's fixed in #714), the code color is still white and I can't `go to definition`. (It says "Can't find any definition of "func")

**Comment by erictraut** [comment]
Thanks for the bug report. This turns out to be a duplicate of a bug that was reported against pyright. It was fixed a couple of days ago and will appear in the next release of pylance.

**Comment by erictraut** [comment]
We'll keep the issue open until the new version is released. Thanks!

**Comment by jakebailey** [comment]
This issue has been fixed in version 2021.1.1, which we've just released. You can find the changelog here: https://github.com/microsoft/pylance-release/blob/main/CHANGELOG.md#202111-13-january-2021
